addons/account/tests/test_sequence_mixin.py

```diff
@@ -437,6 +437,31 @@ def test_journal_override_sequence_regex(self):
         next_move.action_post()
         self.assertEqual(next_move.name, '00000001-G 0002/2017')
 
+    def test_journal_override_sequence_regex_year(self):
+        """Override the sequence regex with a year syntax not matching the draft invoice name"""
+        move = self.create_move(date='2020-01-01')
+        move.journal_id.sequence_override_regex = r'^(?P<prefix1>.*?)?(?P<year>\d{4})(?P<prefix2>.*?)(?P<seq>\d+)$'
```

Comment:
Your regex should always work for any given string, which is not the case here: you require at least `\d{4}` for `year` and `\d+` for `seq`

Here is a suggestion to fix it, by adding a conditional (`?`) on the `year` and the `prefix2` so that they are no required anymore, but doing that required to do a look-behind as well
```suggestion
        move.journal_id.sequence_override_regex = (
            '^'
            r'(?P<prefix1>.*?)'
            r'(?P<year>(?:(?<=\D)|(?<=^))\d{4})?'
            r'(?P<prefix2>(?<=\d{4}).*?)?'
            r'(?P<seq>\d{0,9})'
            r'(?P<suffix>\D*?)'
            '$'
        )
```


If you do this, you'll notice you don't need to change anything in `addons/account/models/sequence_mixin.py`

Reply:
Wouldn't it make things easier if we do not require that an invalid sequence has to match the override regex? This requirement is also not well documented nor the error message is very clear if you provide a regex not satisfying this requirement.
As this PR proves, the changes needed to drop such a requirement on the override regex are very small.

Reply:
Note that also the default `_sequence_monthly_regex` and `_sequence_yearly_regex` do not match an empty string.

Reply:
They don't because they don't need it, only the last one in line needs to match empty lines, so `_sequence_fixed_regex`

Reply:
It's somewhat counterintuitive to me, as someone may start from `_sequence_monthly_regex` to create an override regex and find out it actually doesn't work. Nevertheless, I included your requested changes.

Reply:
Perfect, but then can you also
* revert the changes in `addons/account/models/sequence_mixin.py`
* Keep only 2 commits: one for the fix, and one for the CLA

Reply:
Would you accept a PR against master to improve the override regex support by dropping the (undocumented) requirement that it should match any string?